### ISSUE TITLE
[FEA] Add support for reading DICOM Whole Slide Microscopy Images

### ISSUE THREAD

**Issue by dclunie** [rapidsai/cucim#463]
DICOM WSI is being encouraged for use as a standard format for WSI in order to leverage existing clinical and research archives and to propagate identifying and descriptive metadata (subject, specimen preparation, etc.).

Need cuCIM to be able to read (set of) DICOM files that correspond to a tiled pyramidal encoding +/- label and overview in same manner as it can read SVS, generic TIFF, etc., to take advantage of cuCIM use cases and performance.

BioFormats has already been extended with DICOM WSI support, and addition to OpenSlide is in progress.

DICOM WSI formal documentation is at:
- https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_A.32.8.html

DICOM WSI samples are at:
- ftp://medical.nema.org//MEDICAL/Dicom/DataSets/WG26/ (variable and and quality)
- http://portal.imaging.datacommons.cancer.gov/ (mostly converted from SVS)

**Comment by fedorov** [comment]
> addition to OpenSlide is in progress

As a recent development, integration of a library that will support DICOM read from OpenSlide has been completed https://github.com/openslide/openslide/pull/431, as the first step towards DICOM-enabling OpenSlide.

**Comment by JHancox** [comment]
Thanks @dclunie and @fedorov for your thoughts and information. This has reignited some slow-burning conversations.
One consideration here is that MONAI is the go-to platform for medical imaging, including Digital Pathology, and it may make sense for the DICOM layer to sit within MONAI's somewhat higher level abstractions. For example, cuCIM and OpenSlide are both used within some of MONAI's pathology-specific data loaders, depending on the WSI/patch format. The user does not need to choose which. MONAI tends to use cuCIM whenever possible but defer to OpenSlide for formats for which cuCIM lacks support. This, of course, means that when OpenSlide does support DICOM for pathology, then MONAI will automatically be able to take advantage of this - which is good news. That said, we are open to discussion regarding the most appropriate place for DICOM to be implemented and why cuCIM might be preferred.

**Comment by fedorov** [comment]
Thank you for the update, I will try to remember to update this issue when OpenSlide support of DICOM is fully executed.

**Comment by deepakanand14** [comment]
The OpenSlide library is now supporting DICOM.

**Comment by fedorov** [comment]
Indeed, sorry I completely forgot to update this issue. For the sake of completeness, here is the announcement post by @bgilbert: https://forum.image.sc/t/openslide-4-0-0-released/87323 and the release notes: https://github.com/openslide/openslide/releases/tag/v4.0.0.
